server/src/main/java/com/hedera/block/server/producer/ProducerBlockItemObserver.java

```diff
@@ -108,28 +115,59 @@ public void onNext(@NonNull final PublishStreamRequest publishStreamRequest) {
             @NonNull final var errorResponse = buildErrorStreamResponse();
             publishStreamResponseObserver.onNext(errorResponse);
             LOGGER.log(System.Logger.Level.ERROR, "Exception thrown publishing BlockItem: ", io);
-
             LOGGER.log(System.Logger.Level.ERROR, "Shutting down the web server");
             serviceStatus.stopWebServer();
+        } catch (ParseException e) {
+            @NonNull final var errorResponse = buildErrorStreamResponse();
+            publishStreamResponseObserver.onNext(errorResponse);
+            LOGGER.log(
+                    System.Logger.Level.ERROR,
+                    "Error parsing inbound block item from a producer: "
+                            + publishStreamRequest.getBlockItem(),
+                    e);
+            serviceStatus.stopWebServer();
         }
     }
 
     @NonNull
-    private PublishStreamResponse buildSuccessStreamResponse(@NonNull final BlockItem blockItem)
-            throws IOException, NoSuchAlgorithmException {
-        @NonNull final ItemAcknowledgement itemAck = itemAckBuilder.buildAck(blockItem);
-        return PublishStreamResponse.newBuilder().setAcknowledgement(itemAck).build();
+    private com.hedera.hapi.block.protoc.PublishStreamResponse buildSuccessStreamResponse(
+            @NonNull final BlockItem blockItem) throws IOException, NoSuchAlgorithmException {
+
+        @NonNull final Acknowledgement ack = buildAck(blockItem);
+
+        return toProtocPublishStreamResponse(
+                PublishStreamResponse.newBuilder().acknowledgement(ack).build());
     }
 
     @NonNull
-    private static PublishStreamResponse buildErrorStreamResponse() {
+    private static com.hedera.hapi.block.protoc.PublishStreamResponse buildErrorStreamResponse() {
         // TODO: Replace this with a real error enum.
         @NonNull
         final EndOfStream endOfStream =
                 EndOfStream.newBuilder()
-                        .setStatus(PublishStreamResponseCode.STREAM_ITEMS_UNKNOWN)
+                        .status(PublishStreamResponseCode.STREAM_ITEMS_UNKNOWN)
                         .build();
-        return PublishStreamResponse.newBuilder().setStatus(endOfStream).build();
+        return toProtocPublishStreamResponse(
+                PublishStreamResponse.newBuilder().status(endOfStream).build());
+    }
+
+    /**
+     * Protected method meant for testing. Builds an Acknowledgement for the block item.
```

Comment:
why is this meant for testing? what kind of tests?
 I don't see it referenced in any test, hence my question.

Also, is this something temporary? or expected to be here always.

Reply:
This method is overridden in `TestProducerBlockItemObserver` which is a private class at the bottom of `ProducerBlockItemObserverTest`. It's used to throw an exception.

Reply:
It makes sense, the method is not meant for testing, but is protected so it can be tested. 👍